constellations/gemini-user/src/class/class.js

```diff
@@ -0,0 +1,219 @@
+"use strict";
+require("./class.sass");
+import React from "react";
+import moment from "moment";
+import axios from "axios";
+import API from "../utils/api.js";
+import { Link } from "react-router-dom";
+
+export class ClassPage extends React.Component {
+    state = {
+        classObj: {},
+        sessions: [],
+        program: {},
+        semester: {},
+        location: {},
+    };
+
+    componentDidMount = () => {
+        const classId = this.props.classId;
+        const apiClassCalls = [
+            API.get("api/classes/class/" + classId),
+            API.get("api/sessions/class/" + classId),
+        ];
+
+        let classObj;
+        let sessions;
+        axios
+            .all(apiClassCalls)
+            .then(
+                axios.spread((...responses) => {
+                    classObj = responses[0].data;
+                    sessions = _.sortBy(responses[1].data, ["startsAt"]);
+
+                    const apiCalls = [
+                        API.get("api/programs/program/" + classObj.programId),
+                        API.get(
+                            "api/semesters/semester/" + classObj.semesterId
+                        ),
+                        API.get(
+                            "api/locations/location/" + classObj.locationId
+                        ),
+                    ];
+                    axios
+                        .all(apiCalls)
+                        .then(
+                            axios.spread((...responses) =>
+                                this.setState({
+                                    classObj: classObj,
+                                    sessions: sessions,
+                                    program: responses[0].data,
+                                    semester: responses[1].data,
+                                    location: responses[2].data,
+                                })
+                            )
+                        )
+                        .catch((err) =>
+                            console.log(
+                                "Error: could not fetch program, semester, and location: " +
+                                    err.message
+                            )
+                        );
+                })
+            )
+            .catch(
+                (err) =>
+                    console.log("Error: could not fetch class: " + err.message)
+                // TODO: error page?
+            );
```

Comment:
This looks a bit crazy (especially with the indents), anyway we can modularize these into separate functions? 

Maybe something like:

```
fetchClassAndSessions(apiClassCalls)
    .then( axios.spread((...responses) => {
        fetchOtherData(responses)
            .then(...)
            .catch(...);
    })
    .catch((err) => ...);

fetchOtherData = (responses) => {
   ... get the programs, semesters, and locations ...
}
```

^ something like above. You don't have to do it completely that way. I'm just trying to provide an alternative with how you can organize better. With above, it's clear what the API calling order is. First we get Classes&Sessions, then we get other random stuff based on those responses.
Also, it's decently clear what will happen when failures occur. You can see which `catch` is for which set of API calls.